### PR TITLE
Fix permission name - not show, but view

### PR DESCRIPTION
**Pull request by axzx**
I am targeting this branch, because BC.

## Changelog

```markdown
### Fixed
- name of permission, use `VIEW` instead of `SHOW`
```

## Subject
In `isGranted` param should be name of permission, not action name.

similar PR #4360

### CODE REVIEW

File: Controller/CRUDController.php [hunk 1]
```diff
@@ -1080,7 +1080,9 @@ protected function redirectTo($object)
 
         if (!$url) {
             foreach (array('edit', 'show') as $route) {
-                if ($this->admin->hasRoute($route) && $this->admin->isGranted(strtoupper($route), $object)) {
+                if ($this->admin->hasRoute($route) 
+                    && $this->admin->isGranted(strtoupper(str_replace('show', 'view', $route)), $object)
```
Comment (greg0ire) [inline Controller/CRUDController.php]: I think the `checkAccess` method has built-in logic that should work here. This looks too fragile.

Comment (axzx) [inline Controller/CRUDController.php]: I check this and results is:
```
$this->admin->isGranted('VIEW'); // return true
$this->admin->isGranted('SHOW'); // return false
```

Comment (greg0ire) [inline Controller/CRUDController.php]: I did write `checkAccess`, not `isGranted`

Comment (jordisala1991) [inline Controller/CRUDController.php]: In fact, you might want to do `hasAccess`:

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Admin/AbstractAdmin.php#L2833

`checkAccess` will throw an exception, and that's not what we want here

Comment (axzx) [inline Controller/CRUDController.php]: I don't understand. 
Where is `checkAccess` in this method?;)

Here we have `$this->admin->isGranted`, not `checkAccess`.

Comment (jordisala1991) [inline Controller/CRUDController.php]: Yes, but the correct code would be:

```
$this->admin->hassAccess($route, $object)
```

Comment (greg0ire) [inline Controller/CRUDController.php]: You have access to an action, because you are granted a role, if it helps.

Comment (axzx) [inline Controller/CRUDController.php]: @jordisala1991 I' will create new pull request with this change.

Comment (greg0ire) [inline Controller/CRUDController.php]: Don't

Comment (greg0ire) [inline Controller/CRUDController.php]: Please amend your commit, force push, and edit this PR.

Comment (axzx) [inline Controller/CRUDController.php]: sorry, too late:[

Comment (greg0ire) [inline Controller/CRUDController.php]: argh

**Pull request closed without merging**